### PR TITLE
The Greatest Master Merge.

### PR DESCRIPTION
**Pull request by Tysyatsky**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @Kasterov 
- [ ] @Tysyatsky 
- [ ] @Katerix 
- [ ] @Kotusyk 
- [ ] @NadiaKishchuk 
- [ ] @MementoMorj 
- [ ] @Chynchenko 

### Second Level Review

- [ ] @LanchevychMaxym 
- [ ] @oshut

## Summary of issue

Update the master branch and use github flow properly.

## Summary of change

3 month of work of Streetcode team.

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: src/app/api/partners/partners.api.ts [hunk 1]
```diff
@@ -1,19 +1,24 @@
 import Agent from '@api/agent.api';
 import { API_ROUTES } from '@constants/api-routes.constants';
-import Partner from '@models/partners/partners.model';
+import Partner, { PartnerShort } from '@models/partners/partners.model';
+
+// eslint-disable-next-line no-restricted-imports
+import { PartnerCreateUpdate } from '../../../models/partners/partners.model';
```
Comment (Katerix) [inline src/app/api/partners/partners.api.ts:7]: maybe this needs modification

File: src/app/common/components/ScrollToTopBtn/ScrollToTopBtn.styles.scss [hunk 2]
```diff
@@ -1,9 +1,27 @@
...
+  @include mut.flex-centered();
+
+  @include mut.circular(62px);
+  z-index: 4;
+
+  border: 3px solid c.$accented-red-color;
+  @include vnd.vendored(animation, 'btnOpacity 1s ease-in-out');
+  @include vnd.vendored(transition, 'filter .5s ease');
+  tra
+
```
Comment (Katerix) [inline src/app/common/components/ScrollToTopBtn/ScrollToTopBtn.styles.scss]: cer

File: src/app/common/components/modals/ConfirmationModal/ConfirmationModalstyle..scss
Comment (Katerix) [inline src/app/common/components/modals/ConfirmationModal/ConfirmationModalstyle..scss:1]: needs to be removed

File: src/app/common/components/modals/ListenText/ListenText.styles.scss [hunk 3]
```diff
@@ -38,4 +38,33 @@
...
+  
+}
+@media screen and (max-width: 1024px){
+    .closeModal{
+        display:none;
+    }
+    .buttonContainer {
+        margin-right: f.pxToRem(5px);
+      }
 }
```
Comment (Katerix) [inline src/app/common/components/modals/ListenText/ListenText.styles.scss:70]: this file needs extra spaces

File: src/app/common/components/modals/Tags/TagsModal.component.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import useMobx from '@stores/root-store';
+
+import { Modal } from 'antd';
+
+import { useAsync } from '@/app/common/hooks/stateful/useAsync.hook';
+import RelatedFigureItem
+    from '@/features/StreetcodePage/RelatedFiguresBlock/RelatedFigureItem/RelatedFigureItem.component';
+
+import TagsSliderModal from './TagsSliderModal/TagsSliderModal.component';
+
```
Comment (Katerix) [inline src/app/common/components/modals/Tags/TagsModal.component.tsx:16]: imports without empty lines

File: src/app/common/components/modals/Tags/TagsModal.styles.scss [hunk 5]
```diff
@@ -0,0 +1,77 @@
+@use "@assets/sass/_utils.functions.scss" as *;
+@use "@assets/sass/variables/_variables.fonts.scss" as ft;
+@use "@assets/sass/variables/_variables.colors.scss" as c;
+@use "@assets/sass/mixins/_utils.mixins.scss" as mut;
+@use "@assets/sass/_utils.functions.scss" as fnc;
+@use "@assets/sass/mixins/_vendor.mixins.scss" as vnd;
+@use "@sass/_utils.functions.scss" as f;
+
+
```
Comment (Katerix) [inline src/app/common/components/modals/Tags/TagsModal.styles.scss:9]: remove /n

**Review by Katerix — COMMENTED** [review]
reviewed first 53 files (out of 280). some spacings are needed. I will modify reviewed files. Please continue your review from Term-files.

File: src/app/common/components/Video/Video.component.tsx [hunk 6]
```diff
@@ -1,22 +1,31 @@
...
-            />
+    if (windowSize.width >= 1350) {
+        width = '45vw';
+    }
+    return (
+        <div className="videoComponent">
+            <div className="player-wrapper">
+                <ReactPlayer
+                    width="100%"
+                    height={width}
```
Comment (NadiaKishchuk) [inline src/app/common/components/Video/Video.component.tsx:21]: name variables with understandable names. It is not ok to set height property width variable value

File: src/app/stores/related-terms-store.ts [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    get getRelatedTermsArray() {
+        return Array.from(this.storage.values());
+    }
+
+    public fetchRelatedTermsByTermId = async (id: number) => {
+        try {
+            this.setRelatedTermMap = await relatedTermApi.getAllByTermId(id);
+            // console.log(this.storage.values());
```
Comment (NadiaKishchuk) [inline src/app/stores/related-terms-store.ts:31]: remove this comment

File: src/app/stores/streetcodes-catalog-store.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    constructor() {
+        makeAutoObservable(this);
+    }
+
+    public fetchStreetcodes = async () => {
+        try {
+            const array = await StreetcodesApi.getAll(
+                { Page: null, Amount: null, Sort: null, Title: null, Filter: null } as GetAllStreetcodesRequest,
+            );
+            console.log(array);
```
Comment (NadiaKishchuk) [inline src/app/stores/streetcodes-catalog-store.ts:19]: unnecessary console

File: src/features/AdminPage/NewStreetcode/ArtGallery/ArtGallery.styles.scss [hunk 9]
```diff
@@ -0,0 +1,20 @@
+@use "@sass/mixins/_utils.mixins.scss" as mut;
+@use "@assets/sass/variables/_variables.fonts.scss" as ft;
+
+
```
Comment (NadiaKishchuk) [inline src/features/AdminPage/NewStreetcode/ArtGallery/ArtGallery.styles.scss:4]: spacing

File: src/features/AdminPage/NewStreetcode/ArtGallery/components/ArtGallery/ArtGalleryAdminStyles.styles.scss [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+  .medium{
+    width : 250px; //380
+    height: 353px;//535
+
+  }
+  .large{
+    width: 531px;//805
+    height: 353px;//535
+  }
+
```
Comment (NadiaKishchuk) [inline src/features/AdminPage/NewStreetcode/ArtGallery/components/ArtGallery/ArtGalleryAdminStyles.styles.scss:74]: what is the comment ? and don`t we need use f.pxToRem()?

File: src/features/AdminPage/NewStreetcode/TimelineBlock/NewTimelineModal/NewTimelineModal.component.tsx [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+            timelineItemStore.addTimeline(newTimeline);
+        }
+
+        setIsModalOpen(false);
+        form.resetFields();
+    };
+    const onContextSelect = (value:string) => {
+        const index = historicalContextStore.historicalContextArray.findIndex((c) => c.title === value);
+        if (index < 0) {
+            console.log(value.length);
```
Comment (NadiaKishchuk) [inline src/features/AdminPage/NewStreetcode/TimelineBlock/NewTimelineModal/NewTimelineModal.component.tsx:65]: console.log

File: src/features/AdminPage/StreetcodesTable/StreetcodesTable.component.tsx [hunk 12]
```diff
@@ -0,0 +1,188 @@
...
+                      onClick: () => window.open(`${FRONTEND_ROUTES.STREETCODE.BASE}/${record.index}`,'_blank')
+                    }
+                  }}
+                />
+            </div>
+            <div>
+                <div className='underTableZone'>
+                    <div className='underTableElement'>
+                        <InputNumber className='pageAmountElement' min={1} max={20} defaultValue={1} onChange={(value: any) => {
+                            console.log(value);
```
Comment (NadiaKishchuk) [inline src/features/AdminPage/StreetcodesTable/StreetcodesTable.component.tsx:172]: remove console

**Review by Tysyatsky — COMMENTED** [review]
Review is done, small fixes were made.

File: src/features/StreetcodePage/MapBlock/CheckBox/CheckBox.styles.scss [hunk 13]
```diff
@@ -4,37 +4,67 @@
...
+  top: 195px;
+  left: 50%;
+  transform: translateX(-50%);
   @include mut.flexed(row, center, center);
-  z-index: 2;
+
+  z-index: 3;
   background: c.$pure-white-color;
-  border: 3px solid #787878;
+  // border: 3px solid #787878;
```
Comment (NadiaKishchuk) [inline src/features/StreetcodePage/MapBlock/CheckBox/CheckBox.styles.scss:16]: cooment

File: src/features/StreetcodePage/MapBlock/MapBlock.component.tsx [hunk 14]
```diff
@@ -1,39 +1,33 @@
...
-    // const toponyms = useAsync(() => ToponymsApi
-    //     .getByStreetcodeId(id), [id]).value as Toponym[];
+import MapOSM from './Map/Map.component';
 
+const MapBlock = () => {
+    const { streetcodeStore: { getStreetCodeId } } = useMobx();
     const toponyms = useAsync(() => ToponymsApi
         .getAll(), []).value as Toponym[];
-
-    // console.log(toponyms);
```
Comment (NadiaKishchuk) [inline src/features/StreetcodePage/MapBlock/MapBlock.component.tsx:25]: and this

File: src/features/StreetcodePage/MapBlock/MapBlock.styles.scss [hunk 15]
```diff
@@ -3,10 +3,12 @@
...
+  @include mut.rem-margined($top: 170px);
+   @include mut.positioned-as(relative);
+  .leaflet-container {
     z-index: 1;
   }
-}
\ No newline at end of file
+ 
+  
+}
```
Comment (NadiaKishchuk) [inline src/features/StreetcodePage/MapBlock/MapBlock.styles.scss:14]: spacing

File: src/features/StreetcodePage/MapBlock/Statistics/Info.component.tsx [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+                className="infoContainer"
+                placement="bottomLeft"
+                content={content}
+                trigger="click"
+                open={clicked}
+                onOpenChange={handleClickChange}
+            >
+                <Button
+                    className="infoButton"
+                    icon={<InfoCircleOutlined className="infoIcon" style={{ fontSize: '325%' }} />}
```
Comment (NadiaKishchuk) [inline src/features/StreetcodePage/MapBlock/Statistics/Info.component.tsx:63]: can we set this inline style in the scss file?

File: src/features/StreetcodePage/MapBlock/Statistics/Info.styles.scss [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+  }
+  .infoButton {
+  background-color: transparent;
+  border: none;
+  color: #ff4d4f;
+  
+  &:hover {
+    background-color: #ff4d4f;
+    color: #fff;
+  }
```
Comment (NadiaKishchuk) [inline src/features/StreetcodePage/MapBlock/Statistics/Info.styles.scss:26]: try use our color variables

**Review by NadiaKishchuk — APPROVED** [review]

**Pull request merged**